pyprep/prep_pipeline.py

```diff
@@ -53,20 +83,50 @@ class PrepPipeline:
 
     def __init__(self, raw, prep_params, montage, ransac=True, random_state=None):
         """Initialize PREP class."""
-        self.raw = raw.copy()
-        self.ch_names = self.raw.ch_names
-        self.raw.set_montage(montage)
-        self.raw.pick_types(eeg=True, eog=False, meg=False)
-        self.ch_names_eeg = self.raw.ch_names
-        self.EEG_raw = self.raw.get_data() * 1e6
+        self.raw_eeg = raw.copy()
+
+        # split eeg and non eeg channels
+        self.ch_names_all = raw.ch_names.copy()
+        self.ch_types_all = raw.get_channel_types()
+        self.ch_names_eeg = [
+            self.ch_names_all[i]
+            for i in range(len(self.ch_names_all))
+            if self.ch_types_all[i] == "eeg"
+        ]
+        self.ch_names_non_eeg = list(set(self.ch_names_all) - set(self.ch_names_eeg))
+        self.raw_eeg.pick_channels(self.ch_names_eeg)
+        if self.ch_names_non_eeg == []:
+            self.raw_non_eeg = None
+        else:
+            self.raw_non_eeg = raw.copy()
+            self.raw_non_eeg.pick_channels(self.ch_names_non_eeg)
+
+        self.raw_eeg.set_montage(montage)
+        # raw_non_eeg may not be compatible with the montage
+        # so it is not set for that object
+
+        self.EEG_raw = self.raw_eeg.get_data() * 1e6
         self.prep_params = prep_params
-        self.sfreq = self.raw.info["sfreq"]
+        if self.prep_params["ref_chs"] == "eeg":
+            self.prep_params["ref_chs"] = self.ch_names_eeg
+        if self.prep_params["reref_chs"] == "eeg":
+            self.prep_params["reref_chs"] = self.ch_names_eeg
+        self.sfreq = self.raw_eeg.info["sfreq"]
         self.ransac = ransac
         self.random_state = check_random_state(random_state)
 
+    @property
```

Comment:
I have never seen this `property` decorator ... I'll have to look into what it actually does. But it seems fine. :)

Reply:
@christian-oreilly introduced it in the first commit. I have myself seldomly used decorators.  I think that for this particular purpose it adapts well.

Reply:
@christian-oreilly could you please explain why you added `@property` here?

Reply:
in his last comment in #33 he mentioned that : 

> I did not do it in the .raw object directly in case it can have side effect. I made it available through a property that add on the fly the missing channels to the .raw object and return a copy.

If we implement it as a normal attribute I guess said attribute should be set both in the init and at the end of the fit method.

In a way the current implementation does that since it grabs the current raw_eeg and raw_non_eeg .